src/types.ts

```diff
@@ -37,6 +37,17 @@ export interface UnauthenticatedPageProps {
   setSessionUser: SetSessionUser;
 }
 
+export interface GlobalState {
+  sessionUser: User | null;
+  hasSessionLoaded: boolean;
+}
+
+export interface Action {
```

Comment:
Here, I'd recommend making our Action type more generic - something like 
```ts
{
	type: ACTION_TYPE; // enum of all our defined actions
	payload: any; // or we could do an OR type of all the types of our actions
}

Reply:
I was wondering about that, I was just adding optional properties to Action. Or seems nicer, I think it's a nice check to have.

Reply:
Have you seen the OR thing work? I'm getting errors in the reducer, even with a `!`, e.g.

```
hasSessionLoaded = action.payload.hasSessionLoaded!
```
still gives `Property 'hasSessionLoaded' does not exist on type '{ user: User | null; }'.ts(2339)`.

Reply:
Without the OR thing, individual types for optional variables seems like a nice way to maintain some type-checking?

Reply:
hmm, I was going off of benchling's `client/redux-stores/Action.ts`, which does that, but maybe there are nuances to getting it to work properly that I'm not understanding. what we have is fine for now, and we can figure it out later if we need, but it seems nice to be able to get that working 🤷‍♀

Reply:
Hm ok. Casting or commenting w/ a stronger ts-ignore would both work, but that's a lot of extra typing.

Reply:
I wonder how Benchling has it working

Reply:
ya, let's figure it out later - just getting redux here is a huge advantage!